### PR TITLE
Environment checker improvement

### PR DESCRIPTION
**Pull request by RazTamir**
- Accurate compare was needed
 - Removed DeepDiff and pprint
 - Fixed new_project fixture

### CODE REVIEW

**Comment by RazTamir** [comment]
Verified:

E           ocs_ci.ocs.exceptions.ResourceLeftoversException: 
E           There are leftovers in the environment after test case:
E           Resources added:
E           - '***storageClasses***':
E             - apiVersion: storage.k8s.io/v1
E               kind: StorageClass
E               metadata:
E                 creationTimestamp: '2019-07-25T06:59:56Z'
E                 name: '321'
E                 resourceVersion: '311446'
E                 selfLink: /apis/storage.k8s.io/v1/storageclasses/321
E                 uid: cf9c6ef5-aea9-11e9-8a3f-06b2f99b1fd8
E               provisioner: kubernetes.io/no-provisioner
E               reclaimPolicy: Delete
E               volumeBindingMode: WaitForFirstConsumer
E           
E           Resources removed:
E            - '***pods***':
E             - apiVersion: v1
E               kind: Pod
E               metadata:
E                 annotations:
E                   k8s.v1.cni.cncf.io/networks-status: "[{\n    \"name\": \"openshift-sdn\",\n\
E                     \    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.131.0.20\"\n\
E                     \    ],\n    \"default\": true,\n    \"dns\": {}\n}]"
E                   openshift.io/scc: anyuid
E                 creationTimestamp: '2019-07-25T06:16:13Z'
E                 labels:
E                   app: hello-openshift
E                 name: example
E                 namespace: rook-ceph
E                 resourceVersion: '298295'
E                 selfLink: /api/v1/namespaces/rook-ceph/pods/example
E                 uid: b43393b7-aea3-11e9-8a3f-06b2f99b1fd8
E               spec:
E                 containers:
E                 - image: openshift/hello-openshift
E                   imagePullPolicy: Always
E                   name: hello-openshift
E                   ports:
E                   - containerPort: 8080
E                     protocol: TCP
E                   resources: {}
E                   securityContext:
E                     capabilities:
E                       drop:
E                       - MKNOD
E                     procMount: Default
E                   terminationMessagePath: /dev/termination-log
E                   terminationMessagePolicy: File
E                   volumeMounts:
E                   - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
E                     name: default-token-4tx5n
E                     readOnly: true
E                 dnsPolicy: ClusterFirst
E                 enableServiceLinks: true
E                 imagePullSecrets:
E                 - name: default-dockercfg-jgvmk

File: conf/ocs_basic_install.yml [hunk 1]
```diff
@@ -7,4 +7,4 @@ ENV_DATA:
   platform: 'aws'
   worker_replicas: 3  # here you can define diff number of workers
   master_replicas: 3
-  rook_image: "rook/ceph:v1.0.0-256.g01ba50e" # you can overwrite default rook image
+  rook_image: "rook/ceph:master" # you can overwrite default rook image
```
Comment (ebenahar) [inline conf/ocs_basic_install.yml]: Let's not revert to master

Comment (ebenahar) [reply conf/ocs_basic_install.yml]: Dismiss this comment please

Comment (dahorak) [inline conf/ocs_basic_install.yml]: Not sue, if I miss something, how the change of `rook_image` relates to this PR ("Environment checker improvement")?

Comment (RazTamir) [reply conf/ocs_basic_install.yml]: fixed

File: ocs_ci/framework/conf/default_config.yaml [hunk 2]
```diff
@@ -48,4 +48,4 @@ ENV_DATA:
   # cause we don't need to update it each time new 14.x version is released
   # but only once when move to new version like v15.
   ceph_image: 'ceph/daemon-base:latest-nautilus-devel'
-  rook_image: 'rook/ceph:v1.0.0-256.g01ba50e'
+  rook_image: 'rook/ceph:v0.9.0-519.g111610e'
```
Comment (dahorak) [inline ocs_ci/framework/conf/default_config.yaml]: Still I'm not sure, why the `rook_image` is changed in this PR? (here and on the other places)

Comment (RazTamir) [reply ocs_ci/framework/conf/default_config.yaml]: I'll wait for the PR that fixing the version back to 'master' and rebase.
With that said, @ebenahar raised a good point about a core issue in this implementation I need to fix

**Pull request closed without merging**